physical/map.go

```diff
@@ -51,6 +52,40 @@ func (node *Map) Materialize(ctx context.Context, matCtx *MaterializationContext
 	return execution.NewMap(matExprs, materialized, node.Keep), nil
 }
 
+func isVariableNameRecursive(expr Expression, name octosql.VariableName) bool {
+	switch expr := expr.(type) {
+	case *Variable:
+		return expr.Name.Equal(name)
+	case *AliasedExpression:
+		return isVariableNameRecursive(expr.Expr, name)
+	default:
+		return false
+	}
+}
+
+func getOuterName(expr Expression) octosql.VariableName {
+	switch expr := expr.(type) {
+	case *Variable:
+		return expr.Name
+	case *AliasedExpression:
+		return expr.Name
+	default:
+		return octosql.NewVariableName("")
+	}
+}
+
 func (node *Map) Metadata() *metadata.NodeMetadata {
-	return metadata.NewNodeMetadata(node.Source.Metadata().Cardinality())
+	if node.Keep {
+		return metadata.NewNodeMetadata(node.Source.Metadata().Cardinality(), node.Source.Metadata().EventTimeField())
+	}
+
+	eventTimeField := node.Source.Metadata().EventTimeField()
+	var newEventTimeField octosql.VariableName
+	for i := range node.Expressions {
```

Comment:
Since you access node.Expressions[i] twice maybe for i, expr := would be just a bit more optimal

Reply:
Fixed.